src/test/java/com/amazon/dtasdk/v2/serialization/messages/GetUserIdSerializableTest.java

```diff
@@ -0,0 +1,132 @@
+package com.amazon.dtasdk.v2.serialization.messages;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+import com.amazon.dtasdk.v2.serialization.serializer.JacksonSerializer;
+import com.amazon.dtasdk.v2.serialization.serializer.SerializationException;
+//import com.amazon.dtasdk.v2.serialization.messages.InstantAccessRequest;
+
+public class GetUserIdSerializableTest {
+	private JacksonSerializer serializer = new JacksonSerializer();
+    private GetUserIdSerializableRequest request = new GetUserIdSerializableRequest();
+    private GetUserIdSerializableResponse response = new GetUserIdSerializableResponse();
+    
+    @Test
+    public void testSerialize() throws SerializationException {
+        GetUserIdSerializableRequest request = new GetUserIdSerializableRequest();
+        request.setOperation(InstantAccessOperationValue.GETUSERID);
+        request.setInfoField1("nobody@amazon.com");
+        request.setInfoField2("AMZN");
+        request.setInfoField3("1234");
+
+        String requestString = serializer.encode(request);
+
+        assertEquals("{\"operation\":\"GetUserId\",\"infoField1\":\"nobody@amazon.com\",\"infoField2\":\"AMZN\",\"infoField3\":\"1234\"}",
+                requestString);
+    }
+
+    @Test
+    public void testDeserialize() throws SerializationException {
+        String str = "{\"operation\" : \"GetUserId\", \"infoField1\" : \"nobody@amazon.com\",\"infoField2\" : \"AMZN\",\"infoField3\":\"1234\"}";
+        GetUserIdSerializableRequest request = serializer.decode(str, GetUserIdSerializableRequest.class);
+
+        assertEquals(InstantAccessOperationValue.GETUSERID, request.getOperation());
+        assertEquals("nobody@amazon.com", request.getInfoField1());
+        assertEquals("AMZN", request.getInfoField2());
+        assertEquals("1234", request.getInfoField3());
+    }
+    
+    @Test
+    public void testHashCode() {

```

Comment:
1, why there are so many testHashCode and testEquals?
2, again, I would argue that testing HashCode doesn't really make much sense, and for testing equals, you only need to verify the equals method works for two different objects with identical content.